dataplane/src/pipeline.rs

```diff
@@ -0,0 +1,127 @@
+#![deny(
+    unsafe_code,
+    missing_docs,
+    clippy::all,
+    clippy::pedantic,
+    clippy::unwrap_used,
+    clippy::expect_used,
+    clippy::panic
+)]
```

Comment:
Is there a particular reason to have these here? If we want these in general, can we make them repo-wide (or crate-wide)? I think we have some similar stuff in `net` but probably not in `dataplane` yet.